digital_printing/digital_printing/doctype/print_order/print_order.js

```diff
@@ -213,6 +227,38 @@ erpnext.digital_printing.PrintOrder = class PrintOrder extends frappe.ui.form.Co
 			}
 		});
 	}, 1000);
+
+	create_printed_design_item() {
+		let me = this;
+
+		return frappe.call({
+			method: "create_printed_design_item",
+			doc: me.frm.doc,
+			freeze: true,
+			callback: function(r) {
+				if (!r.exc) {
+					frappe.msgprint(__("Printed Design Items created successfully."))
+					me.frm.reload_doc();
+				}
+			}
+		});
+	}
+
+	create_design_item_bom() {
+		let me = this;
+
+		return frappe.call({
+			method: "create_design_item_bom",
```

Comment:
Do not use controller API, because these kinds of APIs rely on data provided by UI. Instead create a new API that loads the Print Order document and calls the same "create_design_item_bom" function